frontend/src/services/AuthService.ts

```diff
@@ -1,56 +1,76 @@
-import router from '@/router';
-import type { CognitoUserSession } from 'amazon-cognito-identity-js';
-import { Auth } from 'aws-amplify';
-import { readonly, ref } from 'vue';
+import router from "@/router";
+import type { CognitoUserSession } from "amazon-cognito-identity-js";
+import { Auth } from "aws-amplify";
+import { readonly, ref } from "vue";
 
-const FAM_LOGIN_USER = 'famLoginUser'
+const FAM_LOGIN_USER = "famLoginUser";
 
 export interface FamLoginUser {
-    username?: string,
-    idpProvider?: string,
-    roles?: string[],
-    authToken?: CognitoUserSession
+  username?: string;
+  idpProvider?: string;
+  roles?: string[];
+  authToken?: CognitoUserSession;
 }
 
 const state = ref({
-    famLoginUser: localStorage.getItem(FAM_LOGIN_USER)?
-                    JSON.parse(localStorage.getItem(FAM_LOGIN_USER) as string) as (FamLoginUser | undefined | null)
-                    : undefined,
-})
+  famLoginUser: localStorage.getItem(FAM_LOGIN_USER)
+    ? (JSON.parse(localStorage.getItem(FAM_LOGIN_USER) as string) as
+        | FamLoginUser
+        | undefined
+        | null)
+    : undefined,
+});
 
 // functions
 
 function isLoggedIn(): boolean {
-    const loggedIn = !!state.value.famLoginUser?.authToken; // TODO check if token expired later?
-    return loggedIn;
+  const loggedIn = !!state.value.famLoginUser?.authToken; // TODO check if token expired later?
+  return loggedIn;
 }
 
 async function login() {
-    /*
+  /*
         See Aws-Amplify documenation:
         https://docs.amplify.aws/lib/auth/social/q/platform/js/
         https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
     */
-    Auth.federatedSignIn();
+  Auth.federatedSignIn();
 }
 
 async function logout() {
-    Auth.signOut()
-    removeFamUser()
-    console.log("User logged out.")
-    router.push('/')
+  Auth.signOut();
+  removeFamUser();
+  console.log("User logged out.");
+
+  // todo: refactor this part, should get these config variables from backend
+  const env = JSON.parse(window.localStorage.getItem("env_data") || "");
+  const cognitoConfig = {
+    domain: env?.fam_cognito_domain.value,
+    region: env?.fam_cognito_region.value,
+    userPoolWebClientId: env?.fam_console_web_client_id.value,
```

Comment:
This "?" optional was used for special reason at aws-exports; I think it is better removing "?" optional if env is not present.
Otherwise looks good to me!